client/post-editor/editor-location/options.jsx

```diff
@@ -0,0 +1,128 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import React from 'react';
+import { connect } from 'react-redux';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FormCheckbox from 'components/forms/form-checkbox';
+import FormTextInput from 'components/forms/form-text-input';
+import Popover from 'components/popover';
+import PostMetadata from 'lib/post-metadata';
+import { updatePostMetadata } from 'state/posts/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPost } from 'state/posts/selectors';
+
+function statusToBoolean( status ) {
+	return 'public' === status;
+}
+
+class EditorLocationOptions extends React.Component {
+	static propTypes = {
+		isSharedPublicly: PropTypes.oneOf( [ 'private', 'public' ] ),
+		label: PropTypes.string,
+	};
+
+	state = {
+		open: false,
+		renderContext: null,
+	};
+
+	onShareChange = event => {
+		this.props.updatePostMetadata( this.props.siteId, this.props.postId, {
+			geo_public: event.target.checked ? 1 : 0,
+		} );
+	};
+
+	onDescriptionChange = event => {
+		this.props.updatePostMetadata( this.props.siteId, this.props.postId, {
+			geo_address: event.target.value,
+		} );
+	};
+
+	open = () => {
+		this.setState( {
+			open: true,
+		} );
+	};
+
+	toggle = () => {
+		this.setState( {
+			open: ! this.state.open,
+		} );
+	};
+
+	setRenderContext = renderContext => {
+		if ( ! renderContext ) {
+			return;
+		}
+
+		this.setState( { renderContext } );
```

Comment:
It's a bug to set state in a ref callback. Either set an instance property on the class, or use the [React.createRef](https://reactjs.org/docs/refs-and-the-dom.html#creating-refs) API.

Reply:
Thanks for the pointer.  This is addressed in 36843b0.